src/Makefile

```diff
@@ -123,6 +134,14 @@ $(BUILDDIR)/%.o: $(SRCDIR)/%.cpp $(SRCDIR)/llvm-version.h $(HEADERS) $(LLVM_CONF
 $(BUILDDIR)/%.dbg.obj: $(SRCDIR)/%.cpp $(SRCDIR)/llvm-version.h $(HEADERS) $(LLVM_CONFIG_ABSOLUTE) | $(BUILDDIR)
 	@$(call PRINT_CC, $(CXX) $(shell $(LLVM_CONFIG_HOST) --cxxflags) $(CPPFLAGS) $(CXXFLAGS) $(DEBUGFLAGS) -c $< -o $@)
 
+# public header rules
+define install_public_header
+TARGET = $(build_includedir)/julia/$(notdir $(1))
+$$(TARGET): $(1)
+	$(INSTALL_F) $$^ $(build_includedir)/julia/
```

Comment:
something will probably need to create this as a directory first, otherwise won't the first header that gets installed end up named `$(build_includedir)/julia` as a file?